File: frontend/src/app/modules/Municipio.ts
import { ID } from "@datorama/akita";
import { Estado } from "./Estado";

export class Municipio{
  id: ID;
  nome: string;
  estado?: Estado;

  constructor(
    id: ID,
    estado: Estado,
    nome: string = '',
  ) {
    this.id = id;
    this.nome = nome;
    this.estado = estado;
  }

}

File: backend/src/main/java/com/backend/BenchMarks/model/Municipio.java
package com.backend.BenchMarks.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;

import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Entity
@Data
@Table(name="municipio")
public class Municipio extends Localidade{

    @NotNull(message = "O campo nome é obrigatório")
    @Size(min = 3, max = 1000, message = "O campo nome deve ter entre 3 e 1000 caracteres")
    @Column(unique = true)
    private String nome;

    @ManyToOne
    @JoinColumn(name = "estado_id", nullable = false)
    private Estado estado;

    
}
